src/gei/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -491,6 +501,12 @@ private void ValidateGHESOptions(MigrateRepoCommandArgs args, bool cloudCredenti
                 {
                     _log.LogInformation("GHES version is 3.8.0 or later, no need to set cloud storage options here, please set in GHES admin UI.");
                 }
+
+                if (args.KeepArchive)
+                {
+                    _log.LogWarning("GHES version is 3.8.0 or later, there is no archive to keep on the user machine.");
+                }
```

Comment:
1. This is good 👍 not sure about the copy, @timrogers can help here 😃 
2. Other than this, another validation can also be added for when `GhesApiUrl` doesn't have a value but `--keep-archive` does. Basically in the else part of `if (args.GhesApiUrl.HasValue())`. We still probably shouldn't throw and just log a warning as you did here with a message like `--keep-archive can only be provided with --ghes-api-url.`

Reply:
```suggestion
                if (args.KeepArchive)
                {
                    _log.LogWarning("Ignoring --keep-archive option because there is no downloaded archive to keep");
                }
```

Reply:
I think we could error in the case where the `--ghes-api-url` is not set and `--keep-archive` is set, because in this case, we know at the very beginning of the command that the combination of options is invalid, whereas in the above case, we only know once we've checked the GHES version.